src/main/java/org/mule/extension/runtime/ExpirationPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.mule.extension.runtime;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A policy around when should a given instances be considered elegible for expiration.
+ * <p/>
+ * Notice that this contract is not directly tied to the instance to be expired itself. It's tied
+ * to the concept of idle based expiration.
+ *
+ * @since 1.0
+ */
+public interface ExpirationPolicy
+{
+
+    /**
+     * Determines if an instance should be expired based on the last time it was used.
```

Comment:
Is not "based on the last time it was used" too restrictive? I mean, I could use number of active instances, system load, etc to say if that instance should be expired